ui/src/app/views/workflow/sidebar/run/hook/workflow.sidebar.run.hook.component.ts

```diff
@@ -64,6 +64,27 @@ export class WorkflowSidebarRunHookComponent implements OnInit {
                 this.loadOutgoingHookDetails();
             }
         });
+        this._workflowEventStore.outgoingHookEvents().subscribe(
+            hr => {
+                if (hr && this.outgoingHook && this.outgoingHook.id === hr.workflow_node_outgoing_hook_id) {
+                    if (this.wr && this.wr.id === hr.workflow_run_id && this.wr.num === hr.num) {
+                        if (!this.outgoingHookRuns) {
+                            this.outgoingHookRuns = new Array<WorkflowNodeOutgoingHookRun>();
+                        }
+                        let found = false;
+                        for (let i = 0; i < this.outgoingHookRuns.length; i++) {
+                            if (this.outgoingHookRuns[i].subnumber === hr.subnumber) {
+                                this.outgoingHookRuns[i] = hr;
+                                found = true;
```

Comment:
missing break ?

Reply:
or you can simply use `.find` function :)

Reply:
done